workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/EditTexts.kt

```diff
@@ -0,0 +1,107 @@
+package com.squareup.workflow1.ui
+
+import android.text.Editable
+import android.text.Selection.getSelectionEnd
+import android.text.Selection.getSelectionStart
+import android.text.TextWatcher
+import android.widget.EditText
+
+/**
+ * Helper for setting the text value of an [EditText] without changing any metadata about the text,
+ * unless [text] includes a selection. If a selection is passed in, then the selection of [editText]
+ * is set to the incoming selection.
+ *
+ * Intended to be used by [LayoutRunner]s for updating [EditText]s from workflow renderings.
+ */
+@WorkflowUiExperimentalApi
+fun updateText(
```

Comment:
Intentionally didn't define these as extension functions, since they are specifically for use with workflows. But open to changing that if people think it makes sense.

Reply:
I'm pro-extension and pro-discoverability. Is there any reason to think they're dangerous for non-workflow uses?

Reply:
I felt more strongly about this yesterday, but I don't remember why. I'll make them extensions.

Reply:
Oh, it might have been because, when the listener is an extension, you see these three options in the autocomplete:
```
fun addTextChangedListener
fun removeTextChangedListener
fun setTextChangedListener
```
and it's not obvious that the last has a completely different listener type than the others.

Reply:
I dunno. If you see those it's because you opted in by using Workflow in the first place. And on top of that you'll get an experimental warning.

Reply:
Maybe you should offer them in an androidx PR. :)

Reply:
:sweatingtowelguy: